core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -242,6 +242,19 @@ default TableStatistics getTableStatistics(ConnectorSession session, ConnectorTa
         return TableStatistics.empty();
     }
 
+    /**
+     * Returns {@code true} when {@link #getTableStatistics} returns more useful (e.g. more accurate) result
+     * for {@code oldTableHandles} than for {@code newTableHandle}.
+     */
+    default boolean degradesTableStatisticsAccuracy(ConnectorSession session, List<ConnectorTableHandle> oldTableHandles, ConnectorTableHandle newTableHandle)
+    {
+        if (getTableStatistics(session, oldTableHandles.get(0), new Constraint(TupleDomain.all())).equals(TableStatistics.empty()) &&
```

Comment:
Should we check that `oldTableHandles` is a singleton list?

Reply:
i intentionally did this this way. this is just a check to make sure the method is implemented when get stats is implemented.
It doesn't have to be implemented when get stats is not implemented, but join pushdown is.